Makefile

```diff
@@ -1,22 +1,28 @@
+TEST_JSON= $(shell find test -name '*.json')
 
-all: validate validate_with_json
+VALIDATE_WDL= $(shell find . -name '*.wdl' ! -path './test/*')
 
-WDL= $(shell find . -name '*.wdl')
+TEST=java -jar $(CROMWELL_JAR) run
 
-VALIDATE=java -jar /app/womtool.jar validate
+VALIDATE=java -jar $(WOMTOOL_JAR) validate
 
-all-tests := $(addsuffix .test, $(basename $(WDL)))
-all-tests-json := $(addsuffix .json, $(all-tests))
+all-tests := $(addsuffix .test, $(TEST_JSON))
 
-validate: $(all-tests)
-validate_with_json : $(all-tests) $(all-tests-json)
+all-validations := $(addsuffix .validate, $(VALIDATE_WDL))
 
+.PHONY: all
+all: test validate
+
+.PHONY: test
 test : $(all-tests)
 
-test_with_json: test $(all-tests-json)
+.PHONY: validate
+validate: $(all-validations)
 
-%.test : %.wdl
-	$(VALIDATE)  $?
+.PHONY: %.wdl.validate
+%.wdl.validate : 
+	$(VALIDATE) $(basename $@)
 
-%.test.json : %.wdl.json
-	$(VALIDATE) $(basename $? .json) -i $?
```

Comment:
I think it's good to have a validation step that is a "full" validation, i.e. that the json supplies all the required values, ifcourse this will be de-facto tested by the test target, but then the error will be less clear. is it difficult to have three levels of testing? (validation w/o json, validation w/ json, testing with json and comparison to expected results?

Reply:
Good point, added validation with jsons before running full tests